Source/SwiftLintFramework/Rules/Style/ExplicitReturnRule.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import SourceKittenFramework
+import SwiftSyntax
+
+public struct ExplicitReturnRule: ConfigurationProviderRule, SubstitutionCorrectableRule, OptInRule {
+    public var configuration = ExplicitReturnConfiguration()
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "explicit_return",
+        name: "Explicit Return",
+        description: "Prefer explicit returns in closures, functions and getters.",
+        kind: .style,
+        nonTriggeringExamples: ExplicitReturnRuleExamples.nonTriggeringExamples,
+        triggeringExamples: ExplicitReturnRuleExamples.triggeringExamples,
+        corrections: ExplicitReturnRuleExamples.corrections
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        return violationOffsets(file: file).map {
+            StyleViolation(
+                ruleDescription: Self.description,
+                severity: configuration.severityConfiguration.severity,
+                location: Location(file: file, byteOffset: $0)
+            )
+        }
+    }
+
+    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        return violationOffsets(file: file).compactMap {
+            file.stringView.byteRangeToNSRange(ByteRange(location: $0, length: 0))
+        }
+    }
+
+    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
+        return (violationRange, "return ")
+    }
+
+    private func violationOffsets(file: SwiftLintFile) -> [ByteCount] {
+        guard let tree = file.syntaxTree else { return [] }
+
+        let visitors: [ViolationReporting] = [
```

Comment:
what do you think about combining these visitors into one? 

my understanding is that this should make the rule faster, as SwiftSyntax still calls all the possible `visitNode` methods even if you don't override a specific one, so combining the visitors should make this faster.